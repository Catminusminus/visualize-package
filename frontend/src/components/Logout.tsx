/**
 * Author of this code work, Yuki Mogi. c FFRI, Inc. 2019
 */

import * as React from 'react'
import Button from '@material-ui/core/Button'
import Tooltip from '@material-ui/core/Tooltip'

interface Props {
  onClick: any
}

const LogoutButton: React.FC<Props> = ({ onClick }) => {
  return (
    <div>
      <Tooltip title="Disable GitHub OAuth" aria-label="logout">
        <Button variant="contained" onClick={onClick} color="primary">
          Deauthorize App
        </Button>
      </Tooltip>
    </div>
  )
}

export default LogoutButton
