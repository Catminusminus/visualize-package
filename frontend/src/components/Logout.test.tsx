/**
 * Author of this code work, Yuki Mogi. c FFRI, Inc. 2019
 */

import * as React from 'react'
import renderer from 'react-test-renderer'
import Logout from './Logout'

it('Snapshot test of Graph', () => {
  const result = renderer.create(<Logout onClick={() => {}} />).toJSON()
  expect(result).toMatchSnapshot()
})
