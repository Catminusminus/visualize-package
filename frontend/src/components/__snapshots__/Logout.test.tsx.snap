// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test of Graph 1`] = `
<div>
  <button
    aria-describedby={null}
    aria-label="logout"
    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    title="Disable GitHub OAuth"
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Deauthorize App
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </button>
</div>
`;
